include/battle/battle_mon.h

```diff
@@ -0,0 +1,132 @@
+#ifndef POKEPLATINUM_BATTLE_MON_H
+#define POKEPLATINUM_BATTLE_MON_H
+
+#include "constants/moves.h"
+#include "constants/pokemon.h"
+#include "constants/string.h"
+
+typedef struct MoveEffectsData {
+    u32 disabledTurns           :3,
+        encoredTurns            :3,
+        chargedTurns            :2,
+        tauntedTurns            :3,
+        protectSuccessTurns     :2,
+        perishSongTurns         :2,
+        rolloutCount            :3,
+        furyCutterCount         :3,
+        stockpileCount          :3,
+        stockpileDefBoosts      :3,
+        stockpileSpDefBoosts    :3,
+        truantFlag              :1,
+        flashFireFlag           :1;
+    
+    u32 battlerIdLockOn     :2,
+        mimickedMoveSlot    :4,
+        battlerIdBinding    :2,
+        battlerIdMeanLook   :2,
+        lastResortCount     :3,
+        magnetRiseTurns     :3,
+        healBlockTurns      :3,
+        embargoTurns        :3,
+        hasHeldItem         :1, // this is specifically for tracking Unburden; 0 if no item was ever held, or if item is suppresed with Knock Off
```

Comment:
long comment. Either put it above the line and split it up (easier if you don't do the bitfields comma thing), or make the name reflect its use better.